### ISSUE TITLE
[UWP] Cannot disable all annotations.

### ISSUE THREAD

**Issue by leoek** [PSPDFKit/react-native#258]
# Details

If the user selects text in the pdf, a context menu appears which lets the use highlight, underline or strike-through that text. 

![Unbenannt](https://user-images.githubusercontent.com/9424038/61467150-6b6eb600-a97b-11e9-8e50-ab1477b03eba.PNG)

I need to disable that menu for some pdfs. (We have some pdfs which shouldn't have annotations.) However there seems to be no option for that.

I guess this would require a mapping of the configuration supplied to the react-native wrapper to this method: https://pspdfkit.com/api/web/PSPDFKit.Configuration.html#editableAnnotationTypes

# Acceptance Criteria

- [ ] All annotation options can be disabled through the configuration supplied to `PSPDFKitView` or with a call (like `setToolbarItems`)

**Comment by leoek** [comment]
How does the integration on windows work? Could this be done without changes to the proprietary Windows Lib?

**Comment by nickwinder** [comment]
@leoek `editableAnnotationTypes` is not mapped from Web to Windows as of yet. Therefore, we'll have to go through the process of adding it there first. I've opened a zendesk ticket for you which will be handled accordingly. 

I will keep this issue open to ensure we implement the wrapper code for react-native-pspdfkit when this feature is available.

**Comment by leoek** [comment]
@nickwinder Thanks for opening the issue!

**Comment by irgendeinich** [comment]
On Android we could expose the `editableAnnotationTypes` which isn't currently exposed. Using this you can disable editing for all types making the document read-only.

**Comment by radazzouz** [comment]
On iOS, we can set `editableAnnotationTypes` to an empty set, like so:

```js
configuration={{
   editableAnnotationTypes: []
}}
```

Or set `enableAnnotationEditing` to false (once #310 is merged), like this:

```js
configuration={{
   enableAnnotationEditing: false
}}
```

**Comment by nickwinder** [comment]
@irgendeinich @radazzouz The issue with `editableAnnotationTypes` is that UWP does not support this currently, thus we were trying to find a comprimise. 
Does Android support `enableAnnotationEditing`?

**Comment by irgendeinich** [comment]
@nickwinder 
Yes Android also supports `enableAnnotationEditing`.

**Comment by irgendeinich** [comment]
I went ahead and created a PR to expose both `enableAnnotationEditing` as well as `editableAnnotationTypes` on Android.

See https://github.com/PSPDFKit/react-native/pull/311

**Comment by nickwinder** [comment]
Awesome. Thanks guys. As `READONLY` mode in UWP only refers to annotations currently. I'll go ahead and add support by reusing `enableAnnotationEditing` to bring further parity across platforms.

**Comment by nickwinder** [comment]
@leoek I've merged an implementation for `enableAnnotationEditing` into `windows-support` branch. 
`PSPDFKitView` now has a prop of `configuration`, the same as iOS and Android. The only supported feature within the configuration is `enableAnnotationEditing` which is `true` by default. 
`Present` now also takes a configuration (the same as iOS and Android), so please note for windows this is a breaking change to the API. 
I've added a catalog example to show how the configuration can be used. 
I'll close this issue now. Feel free to open again if there is an issue or report anything further in another issue.
